### PR TITLE
Remove perf counters

### PR DESCRIPTION
**Pull request by timbussmann**
To be able to target .net standard, we need to replace our usage of performance counters.

There seem to be other, process relates data about memory usage we might use.
Thoughts @ramonsmits @bording 

(this branch needs to be retargeted once csproj-update has been merged)

### CODE REVIEW

File: src/PerformanceTests/Utils/Statistics.cs [hunk 1]
```diff
@@ -131,10 +117,10 @@ public void Dump()
         if (sendTimeWithTx != TimeSpan.Zero)
             LogStats("SendingInsideTX", Convert.ToDouble(NumberOfMessages / 2) / sendTimeWithTx.TotalSeconds, "msg/s", StatsFormatDouble);
 
-        var counterValues = perfCounterValues.ToList();
-        LogStats("PrivateBytes-Min", counterValues.Min() / 1024, "kb", StatsFormatInt);
-        LogStats("PrivateBytes-Max", counterValues.Max() / 1024, "kb", StatsFormatInt);
-        LogStats("PrivateBytes-Avg", counterValues.Average() / 1024, "kb", StatsFormatInt);
+        LogStats("PrivateMemorySize", process.PrivateMemorySize64 / 1024, "kB", StatsFormatInt);
+        LogStats("PeakWorkingSet", process.PeakWorkingSet64 / 1024, "kB", StatsFormatInt);
+        LogStats("PeakPagedMemorySize", process.PeakPagedMemorySize64 / 1024, "kB", StatsFormatInt);
```
Comment (ramonsmits) [inline src/PerformanceTests/Utils/Statistics.cs]: > kB

File: src/PerformanceTests/Utils/Statistics.cs [hunk 2]
```diff
@@ -131,10 +117,10 @@ public void Dump()
         if (sendTimeWithTx != TimeSpan.Zero)
             LogStats("SendingInsideTX", Convert.ToDouble(NumberOfMessages / 2) / sendTimeWithTx.TotalSeconds, "msg/s", StatsFormatDouble);
 
-        var counterValues = perfCounterValues.ToList();
-        LogStats("PrivateBytes-Min", counterValues.Min() / 1024, "kb", StatsFormatInt);
-        LogStats("PrivateBytes-Max", counterValues.Max() / 1024, "kb", StatsFormatInt);
-        LogStats("PrivateBytes-Avg", counterValues.Average() / 1024, "kb", StatsFormatInt);
+        LogStats("PrivateMemorySize", process.PrivateMemorySize64 / 1024, "kB", StatsFormatInt);
+        LogStats("PeakWorkingSet", process.PeakWorkingSet64 / 1024, "kB", StatsFormatInt);
```
Comment (ramonsmits) [inline src/PerformanceTests/Utils/Statistics.cs]: Wow... those were actually present on the process info? Freaking awesome!

File: src/PerformanceTests/Utils/Statistics.cs [hunk 3]
```diff
@@ -76,12 +69,6 @@ public static IDisposable Initialize(Permutation permutation)
     {
         this.permutation = permutation;
         process = Process.GetCurrentProcess();
-        privateBytesCounter = new PerformanceCounter("Process", "Private Bytes", process.ProcessName);
-        perfCountersTimer = new Timer(state =>
```
Comment (ramonsmits) [inline src/PerformanceTests/Utils/Statistics.cs]: Why removed the timer? Would that be replaced with a call to `process.PrivateMemorySize64`?

Comment (timbussmann) [reply src/PerformanceTests/Utils/Statistics.cs]: not sure I understand the comment. Do you suggest to keep the timer to calculate an average of `process.PrivateMemorySize64`?

Comment (ramonsmits) [reply src/PerformanceTests/Utils/Statistics.cs]: That was the existing behavior. I don't know of how much use it is. @dvdstelt What do you think? Shall we just drop this? The peak memory usage is probably already a good indicator as most tests would just run 1 minute.

**Comment by timbussmann** [comment]
rebased on master

**Pull request merged**